### PR TITLE
Update Python Guide and Scala Guide of Keras-Style API Guide

### PR DESCRIPTION
**Pull request by Deliadd**

### CODE REVIEW

**Comment by Deliadd** [comment]
@hkvision Please help me review this pr. Thanks.

File: docs/docs/KerasStyleAPIGuide/keras-api-python.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+```python
+from zoo.pipeline.api.keras.models import *
+from zoo.pipeline.api.keras.layers import *
+```
+
+and making modifications subject to the following limitations:
+
+1. The Keras version we support and test is [__Keras 1.2.2__](https://faroit.github.io/keras-docs/1.2.2/) with TensorFlow backend.
+
+2. There exist some arguments supported in Keras layers but not supported in Analytics Zoo for now. See [here](../../APIGuide/keras-issues/#unsupported-layer-arguments) for the full list of unsupported layer arguments. Also, currently we haven't supported self-defined Keras layers or [`Lambda`](https://faroit.github.io/keras-docs/1.2.2/layers/core/#lambda) layers.
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/keras-api-python.md]: the link is invalid. Link it to bigdl corresponding page
And remove `Also, currently we haven't supported self-defined Keras layers or [`Lambda`](https://faroit.github.io/keras-docs/1.2.2/layers/core/#lambda) layers.`

**Comment by Deliadd** [comment]
Jenkins is successful, ID is 1498.
Check the link here: http://10.239.47.211:18888/view/ZOO-PR/job/ZOO-PR-Validation/1498/

File: docs/docs/KerasStyleAPIGuide/keras-api-python.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+```python
+from zoo.pipeline.api.keras.models import *
+from zoo.pipeline.api.keras.layers import *
+```
+
+and making modifications subject to the following limitations:
+
+1. The Keras version we support and test is [__Keras 1.2.2__](https://faroit.github.io/keras-docs/1.2.2/) with TensorFlow backend.
+
+2. There exist some arguments supported in Keras layers but not supported in Analytics Zoo for now. See [here](https://bigdl-project.github.io/0.7.0/#APIGuide/keras-issues/#unsupported-layer-arguments) for the full list of unsupported layer arguments.
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/keras-api-python.md]: change the link from 0.7.0 to master?

File: docs/docs/KerasStyleAPIGuide/keras-api-scala.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
+## **Introduction**
+We hereby introduce a new set of __Keras-Style API__ based on [__Keras 1.2.2__](https://faroit.github.io/keras-docs/1.2.2/) in Analytics Zoo for the sake of user-friendliness. Users, especially those familiar with Keras, are recommended to use the new API to create an Analytics Zoo model and train, evaluate or tune it in a distributed fashion.
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/keras-api-scala.md]: modify as python does... such as remove new set of...

File: docs/docs/KerasStyleAPIGuide/keras-api-scala.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+To define a model in Scala using the Keras-Style API, now one just need to import the following packages:
+
+```scala
+import com.intel.analytics.zoo.pipeline.api.keras._
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/keras-api-scala.md]: .keras.layers._
.keras.models._???

File: docs/docs/KerasStyleAPIGuide/keras-api-scala.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+getOutputShape()
+```
+* Return the input or output shape of a model, which is a [`Shape`](#shape) object. For `SingleShape`, the first entry is `-1` representing the batch dimension. For a model with multiple inputs or outputs, it will return a `MultiShape`.
+
+```scala
+setName(name)
+```
+* Set the name of the model.
+
+See [here](Optimization/training/) on how to train, predict or evaluate a defined model.
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/keras-api-scala.md]: remove this.

File: docs/docs/KerasStyleAPIGuide/keras-api-scala.md [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+To create a sequential container:
+```scala
+Sequential()
+```
+
+Example code to create a sequential model:
+```scala
+import com.intel.analytics.zoo.pipeline.api.keras.layers.Dense
+import com.intel.analytics.zoo.pipeline.api.keras.layers.Activation
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/keras-api-scala.md]: put together {Dense, Activation}?

File: docs/docs/KerasStyleAPIGuide/keras-api-scala.md [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+Parameters:
+
+* `inputs`: A list of node instances. Must be more than one node.
+* `mode`: Merge mode. String, must be one of: 'sum', 'mul', 'concat', 'ave', 'cos', 'dot', 'max'. Default is 'sum'.
+* `concatAxis`: Int, axis to use when concatenating nodes. Only specify this when merge mode is 'concat'. Default is -1, meaning the last axis of the input.
+
+Example code to create a graph model:
+```scala
+import com.intel.analytics.zoo.pipeline.api.keras.layers.Dense
+import com.intel.analytics.zoo.pipeline.api.keras.layers.Input
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/keras-api-scala.md]: put together

File: docs/docs/KerasStyleAPIGuide/keras-api-scala.md [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+import com.intel.analytics.zoo.pipeline.api.keras.models.Model
+import com.intel.analytics.bigdl.utils.Shape
+import com.intel.analytics.zoo.pipeline.api.keras.Merge.merge
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/keras-api-scala.md]: put this line above Shape import? this will be good looking...

File: docs/docs/KerasStyleAPIGuide/keras-api-scala.md [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+
+To set the name of a layer, call the method `setName(name)` of the layer.
+
+---
+## **LeNet Example**
+Here we adopt our Keras-Style API to define a LeNet CNN model to be trained on the MNIST dataset:
+
+```scala
+import com.intel.analytics.bigdl.numeric.NumericFloat
+import com.intel.analytics.zoo.pipeline.api.keras._
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/keras-api-scala.md]: can this work?

File: docs/docs/KerasStyleAPIGuide/keras-api-scala.md [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+model.add(Convolution2D(12, 5, 5, activation = "tanh").setName("conv2_5x5"))
+model.add(MaxPooling2D())
+model.add(Flatten())
+model.add(Dense(100, activation = "tanh").setName("fc1"))
+model.add(Dense(10, activation = "softmax").setName("fc2"))
+
+model.getInputShape().toSingle().toArray // Array(-1, 28, 28, 1)
+model.getOutputShape().toSingle().toArray // Array(-1, 10)
+```
+See [here](https://github.com/intel-analytics/BigDL/tree/master/spark/dl/src/main/scala/com/intel/analytics/bigdl/example/keras) for detailed introduction of LeNet, the full example code and running instructions.
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/keras-api-scala.md]: remove

File: docs/docs/KerasStyleAPIGuide/keras-api-scala.md [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+Model(input, output)
+```
+Parameters:
+
+* `input`: An input node or an array of input nodes.
+* `output`: An output node or an array of output nodes.
+
+To merge a list of input __nodes__ (__NOT__ layers), following some merge mode in the Functional API:
+```scala
+import com.intel.analytics.zoo.pipeline.api.keras.Merge.merge
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/keras-api-scala.md]: same as above

**Comment by hkvision** [comment]
LGTM besides the two comments above.
